src/CSESoftware.Repository.EntityFramework/Repository.cs

```diff
@@ -13,30 +17,78 @@ public Repository(TContext context) : base(context)
         }
 
         public virtual void Create<TEntity>(TEntity entity)
-            where TEntity : class, IBaseEntity
+            where TEntity : class, IEntity
         {
-            entity.IsActive = true;
-            entity.CreatedDate = DateTime.UtcNow;
+            if (entity is IActiveEntity activeEntity)
+                entity = PrepareService.SetIsActiveToTrue<TEntity>(activeEntity);
+
+            if (entity is IModifiedEntity modifiedEntity)
+                entity = PrepareService.SetCreatedDateToNow<TEntity>(modifiedEntity);
+
             Context.Set<TEntity>().Add(entity);
         }
 
+        public virtual void Create<TEntity>(List<TEntity> entities)
+            where TEntity : class, IEntity
+        {
+            var entitiesToSave = new List<TEntity>();
+
+            foreach (var entity in entities)
+            {
+                var tempEntity = entity;
+
+                if (entity is IActiveEntity activeEntity)
```

Comment:
Could we use a better solution here? Maybe a factory pattern? If blocks like these don't scale well, Geoff may have a solution but if he doesn't im fine with this as is, **for now**